plugins/ui/src/js/src/elements/spectrum/TextField.tsx

```diff
@@ -18,16 +22,36 @@ function TextField(props: DHCTextFieldProps): JSX.Element {
   } = props;
 
   const [value, setValue] = useState(propValue ?? defaultValue);
+  const [pending, setPending] = useState(false);
+  const prevPropValue = usePrevious(propValue);
+
+  if (
+    propValue !== prevPropValue &&
+    propValue !== value &&
+    propValue !== undefined &&
+    !pending
+  ) {
+    setValue(propValue);
+  }
+
+  const propDebouncedOnChange = useCallback(
+    (newValue: string) => {
+      propOnChange(newValue);
+      setPending(false);
+    },
+    [propOnChange]
+  );
```

Comment:
We need to `await` the response from `propOnChange` - otherwise we just call it and we reset pending right away, but we should only reset after the response has been received. Something like
```suggestion
  const propDebouncedOnChange = useCallback(
    async (newValue: string) => {
      try {
        await propOnChange(newValue);
      } catch (e) {
        log.warn('Error returned from onChange', e)
      }
      setPending(false);
    },
    [propOnChange]
  );
```